src/orion/core/utils/backward.py

```diff
@@ -141,7 +141,10 @@ def get_algo_requirements(algorithm):
         )
 
         return requirements
-
+    if isinstance(algorithm, str):
```

Comment:
This should not happen. If there is an issue with the algo plugin discoverability it would need to be fixed at the level of the factory. This fix should be removed so that we can see again the error and fix it in Factory.